core/clingutils/CMakeLists.txt

```diff
@@ -117,6 +117,13 @@ if (runtime_cxxmodules)
   # some systems install it under /usr/include/Vc/Vc which allows rootcling to
   # discover it and assert that the modulemap is not found.
   set(custom_modulemaps ${custom_modulemaps} vc.modulemap)
+
+  # We need to override the default modulemap because instead of producing a
+  # single std.pcm, produces millions of pcms.
+  if (CMAKE_SYSTEM_NAME MATCHES Darwin AND MACOSX_VERSION VERSION_GREATER 14.3)
```

Comment:
Maybe here the conditional could be
```
if (CMAKE_SYSTEM_NAME MATCHES Darwin AND CMAKE_CXX_COMPILER_VERSION VERSION_GREATER_EQUAL 15.0.0.15000309)
```
In order to allow recent versions of the sdk also on macos versions which are not 14.4 but also older.

Reply:
testing the change here https://github.com/root-project/root/pull/14931